### DISCUSSION TITLE
Offline/Online events?

### DISCUSSION THREAD

**Discussion by LukeTOBrien** [dotnet/maui#12112] [Q&A] [answered]
Hello,

I am creating an app and I need to know when the app is online or offline so then I can disable certain buttons when offline and enable them when back online, does MAUI currently have any events for this?

More specifically I am building a Blazor hybrid app, I have actually created some code using the [window.offline](https://developer.mozilla.org/en-US/docs/Web/API/Window/offline_event) and [online](https://developer.mozilla.org/en-US/docs/Web/API/Window/online_event) events, but the listeners for these events don't seem to fire, I've also noticed that `console.log` statements and network traffic don't seem to appear.  

I know the MAUI is brand new and maybe some functionality is still to be implemented.

Thanks

Luke

**Comment by mhrastegari** [comment, marked as answer]
You can monitor for changes in the device's network conditions, check the current network access, and how it is currently connected using MAUI.Essentials.

https://learn.microsoft.com/en-us/xamarin/essentials/connectivity

**Reply by LukeTOBrien** [reply]
Thank you very much!
